### PR TITLE
Fix shader uniform memory leaks

### PR DESCRIPTION
**Pull request by postcasio**
Free memory for int and float array uniforms when the uniform or shader is destroyed.

### CODE REVIEW

File: src/minisphere/galileo.c [hunk 1]
```diff
@@ -409,10 +409,27 @@ shader_ref(shader_t* it)
...
+
 	if (it == NULL || --it->refcount > 0)
 		return;
 
 	console_log(3, "disposing shader program #%u no longer in use", it->id);
+
+	iter = vector_enum(it->uniforms);
+	while ((uniform = iter_next(&iter))) {
+		switch (uniform->type)
+		{
```
Comment (fatcerberus) [inline src/minisphere/galileo.c]: Style: Put the brace on the same line as the switch.

File: src/minisphere/galileo.c [hunk 2]
```diff
@@ -409,10 +409,27 @@ shader_ref(shader_t* it)
 void
 shader_unref(shader_t* it)
 {
+	struct uniform *uniform;
+	iter_t iter;
```
Comment (fatcerberus) [inline src/minisphere/galileo.c]: I wasn't 100% sure about this, but it seems like the prevailing style is to have the space after the `*`, not before.  Also, this is a bit of a weird style quirk, but `iter_t iter;` should be on its own line; I have this odd OCD thing where I refuse to declare variables inline but nonetheless insist of having "loop variables" set off from the rest.

**Review by fatcerberus — CHANGES_REQUESTED** [review]
Just a couple minor style complaints; the change looks right to me otherwise.

File: src/minisphere/galileo.c (hunk 1 above)
Comment (postcasio) [reply src/minisphere/galileo.c]: Done!

File: src/minisphere/galileo.c (hunk 2 above)
Comment (postcasio) [reply src/minisphere/galileo.c]: I changed this anyway to fit with the rest of the code - `*` with the type, and in columns. I forgot to add a blank line before iter - I can add that but I don't want to litter the commit history...

Comment (fatcerberus) [reply src/minisphere/galileo.c]: It's fine, I typically do a squash merge anyway.

**Review by fatcerberus — APPROVED** [review]
Seems good for now. I'll see if I can find a way to get rid of the code duplication some other time Thanks for finding the memory leak, kind of a silly oversight on my part.

**Pull request merged**